zcash_primitives/src/transaction/mod.rs

```diff
@@ -486,7 +494,11 @@ impl Transaction {
             .map_err(|_| io::Error::new(io::ErrorKind::InvalidData, "valueBalance out of range"))
     }
 
-    fn read_v5<R: Read>(mut reader: R, version: TxVersion) -> io::Result<Self> {
+    fn read_v5<R: Read>(
+        mut reader: R,
+        version: TxVersion,
+        consensus_branch_id: BranchId,
+    ) -> io::Result<Self> {
         let (_, lock_time, expiry_height) = Self::read_v5_header_fragment(&mut reader)?;
```

Comment:
Bug: this is overwriting the consensus branch ID committed to in the transaction, with the caller-provided one (which may not be correct, resulting in a different txid).
```suggestion
    fn read_v5<R: Read>(mut reader: R, version: TxVersion) -> io::Result<Self> {
        let (consensus_branch_id, lock_time, expiry_height) =
            Self::read_v5_header_fragment(&mut reader)?;
```